src/rr/rfc_4034.rs

```diff
@@ -116,3 +117,30 @@ impl Display for DS {
         )
     }
 }
+
+#[derive(Debug, PartialEq, Clone, Eq, Hash)]
+pub struct NSEC {
+    pub domain_name: DomainName,
+    pub ttl: u32,
+    pub class: Class,
+    pub next_domain_name: DomainName,
+    pub type_bit_maps: Vec<Type>,
```

Comment:
Change it to a set type like `BTreeSet` as the same types cannot appear multiple time